### DISCUSSION TITLE
Slow execution time.

### DISCUSSION THREAD

**Discussion by mattrienz** [liquidctl/liquidctl#638] [Q&A]
Created a script to run at startup:

cd /D "D:\Utilities\LiquidCTL\venv\Scripts"
call activate.bat
liquidctl initialize --match kraken
liquidctl --match kraken set pump speed  100
liquidctl --match kraken set logo color fixed FF8C00
liquidctl --match kraken set ring color loading FF8C00

The CMD window hands at the initialize line, takes forever to get past it. Once it does, my keyboard and mouse and headset goes black like they were turned off for a second, then they come back on and the Liquid CTL settings apply to my X52. Though, only the logo color was applied, the ring color wasn't applied. Not sure whats going on here. I think its conflicting with ICUE? Can I make liquid CTL only detect the kraken and no other device?

**Comment by Serphentas** [comment]
Thanks for raising this issue. Did you start seeing this behaviour from a specific point in time, after some changes (ex. firmware update), or was it always present ? Are you able to reproduce this on another OS, to rule out any Windows-specific problems ?

**Comment by mattrienz** [comment]
I just got liquidctl, new user, no idea when it started it was always there.On Nov 16, 2023, at 4:39 PM, Serphentas ***@***.***> wrote:﻿
Thanks for raising this issue. Did you start seeing this behaviour from a specific point in time, after some changes (ex. firmware update), or was it always present ? Are you able to reproduce this on another OS, to rule out any Windows-specific problems ?

**Comment by jonasmalacofilho** [comment]
> Not sure whats going on here. I think its conflicting with ICUE?

I suppose that's possible. But a conflict would be more likely with NZXT CAM or, perhaps, other non-Corsair-specific monitoring and fan/RGB control software (e.g. HWiNFO, OpenRGB, etc.).

If you can figure out (by exclusion) that there's indeed a conflict which some other piece of software, we can try to address it or at least document the behavior.

> Can I make liquid CTL only detect the kraken and no other device?

With the commands you're using, liquidctl doesn't generate _any_ USB traffic to any devices other than your Kraken.

(In other situations, like when calling `status` or `initialize all`, or when using `--serial` number filters, it indeed would, but only to a reasonable extent; nothing that should make completely unrelated devices, like your keyboard and mouse, blink).

---

> The CMD window hands at the initialize line, takes forever to get past it. 

And on Discord you mentioned it taking over a minute...

This very long but finite delay plus the blinking of unrelated devices makes me also consider that it might not just be a simple conflict between similar tools. Perhaps a USB hub is resetting for some reason. Or some antivirus software is doing stuff...

**Comment by mattrienz** [comment]
One thing I should also mention I’m running a modded bios for my Z390 Aorus Ultra. I have a funny feeling it could be that. Gonna reflash to official and see.On Nov 19, 2023, at 1:25 AM, Jonas Malaco ***@***.***> wrote:﻿

Not sure whats going on here. I think its conflicting with ICUE?

I suppose that's possible. But a conflict would be more likely with NZXT CAM or, perhaps, other non-Corsair-specific monitoring and fan/RGB control software (e.g. HWiNFO, OpenRGB, etc.).
If you can figure out (by exclusion) that there's indeed a conflict which some other piece of software, we can try to address it or at least document the behavior.

Can I make liquid CTL only detect the kraken and no other device?

With the commands you're using, liquidctl doesn't generate any USB traffic to any devices other than your Kraken.
(In other situations, like when calling status or initialize all, or when using --serial number filters, it indeed would, but only to a reasonable extent; nothing that should make completely unrelated devices, like your keyboard and mouse, blink).


The CMD window hands at the initialize line, takes forever to get past it.

And on Discord you mentioned it taking over a minute...
This very long but finite delay plus the blinking of unrelated devices makes me also consider that it might not just be a simple conflict between similar tools. Perhaps a USB hub is resetting for some reason. Or some antivirus software is doing stuff...

**Comment by jonasmalacofilho** [comment]
Oh, if possible, please don't reply directly from the email notification. If you check out the discussion page on GitHub, you'll see that the results are usually terrible (unless you're quite careful). You also miss any potential edits that might have been made to the messages you're replying to.

**Reply by mattrienz** [reply]
Sorry, yea I think my modded bios with the older more efficient microcode is a bit finicky. Gonna go back to official and see

**Reply by mattrienz** [reply]
Hi there just on a fresh install of windows now, back on the official bios, and im getting a weird error when trying to initalize all: liquidctl initialize all
usb.core.NoBackendError: No backend available

No idea whats going on.

**Reply by mattrienz** [reply]
So the bios didnt fix it, still getting really slow execution times, not sure what to do.
